lib/mdl/doc.rb

```diff
@@ -34,7 +34,7 @@ def initialize(text, ignore_front_matter = false)
       else
         @offset = 0
       end
-      @lines = text.split(/\R/)
+      @lines = text.split(/\R/, -1)
```

Comment:
can we add a comment here?
```suggestion
      # the -1 is to cause split to preserve an extra entry in the array so we can
      # tell if there's a final newline in the file or not.
      @lines = text.split(/\R/, -1)
```